vars/deployPlannerSnapshot.groovy

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/groovy
+
+import java.util.regex.Pattern
+
+def call(body) {
+    // evaluate the body block, and collect configuration into the object
+    def config = [:]
+    body.resolveStrategy = Closure.DELEGATE_FIRST
+    body.delegate = config
+    body()
+
+    def yaml
+    def openShiftProject = config.openShiftProject
+    def openShiftTemplate = config.openShiftTemplate
+    def originalImageName = config.originalImageName
+    def newImageName = config.newImageName
+    def deploymentName = config.githubRepo
+    def providerLabel = config.providerLabel ?: 'fabric8'
+    def project = config.githubProject
+
+    def flow = new io.fabric8.Fabric8Commands()
+    def utils = new io.fabric8.Utils()
+
+    openShiftProject = openShiftProject + '-' + utils.getRepoName()
+    container('clients') {
+        if (!flow.isAuthorCollaborator("", project)){
+            currentBuild.result = 'ABORTED'
+            error 'Change author is not a collaborator on the project, aborting build until we support the [test] comment'
+        }
+
+        yaml = flow.getUrlAsString(openShiftTemplate)
+        yaml = Pattern.compile("- image: ${originalImageName}:(.*)").matcher(yaml).replaceFirst("- image: ${newImageName}")
+
+        if (!yaml.contains(newImageName)){
+            error "original image ${originalImageName} not replaced with ${newImageName} in yaml: \n ${yaml}"
+        }
+    }
+    // cant use writeFile as we have long filename errors
+    sh "echo '${yaml}' > snapshot.yml"
+
+    container('clients') {
+        try {
+            sh "oc get project ${openShiftProject} | grep Active"
+        } catch (err) {
+            echo "${err}"
+            sh "oc new-project ${openShiftProject}"
+        }
+
+        sh "oc process -n ${openShiftProject} -f ./snapshot.yml | oc apply -n ${openShiftProject} -f -"
+
+        sleep 10
+        // ok bad bad but there's a delay between DC's being applied and new pods being started.  lets find a better way to do this looking at teh new DC perhaps?
+
+        waitUntil {
+            // wait until the pods are running has been deleted
+            try {
+                sh "oc get pod -l app=${deploymentName},provider=${providerLabel} -n ${openShiftProject} | grep Running"
+                echo "${deploymentName} pod is running"
+                return true
+            } catch (err) {
+                echo "waiting for ${deploymentName} to be ready..."
+                return false
```

Comment:
Is this `waitUntill` a tight loop? Do you think there should be a sleep if it fails? Is that inbuilt? If we add a sleep before the `return false`, we can remove the `sleep 10` above?